### ISSUE TITLE
BrowserFunction does not work with Content Security Policies

### ISSUE THREAD

**Issue by Lukas-Habring** [eclipse-platform/eclipse.platform.swt#276]
**Describe the bug**
A content security policy can restrict the use of BrowserFunctions.

**To Reproduce**
1) Add a BrowserFunction.
2) Access a webpage with a restrictive content security policy header.
3) Listen for the page-loaded-event in your java-code.
4) After the page is loaded, inject some Javascript via org.eclipse.swt.browser.Browser.evaluate() and call your BrowserFunction.
You get an exception, but if you catch it you can see in the debugger inside the browser that the request was blocked by the CSP.

**Expected behavior**
Always allow BrowserFunctions to be called ignoring the CSP that is currently active.

**Environment:**
1. Select the platform(s) on which the behavior is seen:
- - [ ] All OS
- - [ ] Windows
- - [ X ] Linux
- - [ ] macOS 

2. Additional OS info (e.g. OS version, Linux Desktop, etc)
Fedora 36

3. JRE/JDK version
1.18

**Version since**
4.24

**Comment by mickaelistria** [comment]
> Access a webpage with a restrictive content security policy header.

Do you have a good and simple example of such a page available publicly?

**Comment by Lukas-Habring** [comment]
No, unfortunately I have not. But I invested some time to create a simple test-case for you.
![swt browser issue](https://user-images.githubusercontent.com/102620478/181446954-63d9f997-b086-452d-9e33-6f9d233f9296.png)
[SWTBrowserBug.zip](https://github.com/eclipse-platform/eclipse.platform.swt/files/9206638/SWTBrowserBug.zip)

**Comment by Phillipus** [comment]
I tidied up the Snippet so it runs on all platforms. Fails only on Linux.

[SWTBrowserBug.zip](https://github.com/eclipse-platform/eclipse.platform.swt/files/9226045/SWTBrowserBug.zip)

**Comment by Lukas-Habring** [comment]
@Phillipus Nope, I can confirm that the bug still occurs. Furthermore, I cannot see any difference between your version and mine that is significant regarding this bug.
Your version is just a little bit cleaner, but this is not a programming exercise, it's just a quick-and-dirty demonstration of a bug. And for that it is ok to ignore some coding guidelines in my opinion.

**Comment by Phillipus** [comment]
@Lukas-Habring You asked:

> The bug may also appear on other operating systems. Unfortunately, I don't have a Windows-PC or Mac. Can someone test that?

I did that.

But the snippet throws a NPE on Windows and Mac. So I provided a test snippet that doesn't throw a NPE in case anyone else wanted to test it.

In so doing I can confirm your bug still occurring on Linux, but not on Windows and Mac.

I think next time I'll mind my own business.

Goodbye.

**Comment by Lukas-Habring** [comment]
Ok, that was a misunderstanding then. Thanks for the clarification.
Still: There shouldn't be a NullPointerException when you call shell.dispose() from a BrowserFunction, right?

**Comment by Phillipus** [comment]
> There shouldn't be a NullPointerException when you call shell.dispose() from a BrowserFunction, right?

In snippets such as this, it's best just to dispose the `Display` object when the event loop is finished and follow the basic pattern in other SWT snippets. (That is not a criticism of your snippet.)

However, getting back to the original issue, the important thing for your use case is that I can confirm that the Security Policy issue is only occuring on Linux.
